src/services/entity-type.js

```diff
@@ -18,11 +18,14 @@ module.exports = class EntityHelper {
 	 * @returns {JSON} - Created entity type response.
 	 */
 
-	static async create(bodyData, id, orgId) {
+	static async create(bodyData, id, orgId, roles) {
 		bodyData.created_by = id
 		bodyData.updated_by = id
 		bodyData.organization_id = orgId
 		try {
+			const isAdmin = roles.some((role) => role.title === common.ADMIN_ROLE)
+			bodyData.allow_filtering = isAdmin ? bodyData.allow_filtering : false
```

Comment:
@nevil-mathew if allow_filtering  is present in body then get the role from roles[] then check the role